src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -904,7 +947,17 @@ int dhcp_server_handle_message(sd_dhcp_server *server, DHCPMessage *message,
                                 return r;
                         }
 
-                        lease->expiration = req->lifetime * USEC_PER_SEC + time_now;
+                        expiration = req->lifetime * USEC_PER_SEC + time_now;
+
+                        if (!existing_lease) {
+                                prepare_new_lease(&lease, static_lease->address, &req->client_id,
```

Comment:
```
systemd-networkd[71079]: ../build/src/libsystemd-network/sd-dhcp-server.c:953:33: runtime error: member access within null pointer of type 'struct sd_dhcp_static_lease'
systemd-networkd[71079]:     #0 0x564761df76d0 in dhcp_server_handle_message ../build/src/libsystemd-network/sd-dhcp-server.c:953
systemd-networkd[71079]:     #1 0x564761df87b4 in server_receive_message ../build/src/libsystemd-network/sd-dhcp-server.c:1074
systemd-networkd[71079]:     #2 0x7f9f63cd9426 in source_dispatch ../build/src/libsystemd/sd-event/sd-event.c:3193
systemd-networkd[71079]:     #3 0x7f9f63cdf89e in sd_event_dispatch ../build/src/libsystemd/sd-event/sd-event.c:3634
systemd-networkd[71079]:     #4 0x7f9f63ce0003 in sd_event_run ../build/src/libsystemd/sd-event/sd-event.c:3692
systemd-networkd[71079]:     #5 0x7f9f63ce029a in sd_event_loop ../build/src/libsystemd/sd-event/sd-event.c:3714
systemd-networkd[71079]:     #6 0x564761c4578e in run ../build/src/network/networkd.c:123
systemd-networkd[71079]:     #7 0x564761c4629e in main ../build/src/network/networkd.c:130
systemd-networkd[71079]:     #8 0x7f9f626cf022 in __libc_start_main (/usr/lib/libc.so.6+0x27022)
systemd-networkd[71079]:     #9 0x564761c4502d in _start (/systemd-meson-build/systemd-networkd+0x2dd02d)
systemd-networkd[71079]: SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../build/src/libsystemd-network/sd-dhcp-server.c:953:33 in
```

https://ci.centos.org/job/systemd-pr-build-vagrant-sanitizers/5565/artifact//systemd-centos-ci/artifacts_all/artifacts__LbYa0/vagrant-logs.i9X/vagrant-arch-sanitizers-gcc-testsuite.HkA/check-networkd-log-for-sanitizer-errors_FAIL.log